java/mykola-melnyk/src/com/zoolatech/lecture7/tasks/_2/Calc_v_2.java

```diff
@@ -0,0 +1,33 @@
+package com.zoolatech.lecture7.tasks._2;
+
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class Calc_v_2 implements Calc {
+    private int storedValue;
+    private Lock calcLock = new ReentrantLock();
+    public Calc_v_2(int value) {
+        storedValue = value;
+    }
+    public synchronized int getStoredValue() {
```

Comment:
please replace `synchronized` with a lock

Reply:
Sorry, forgot that this is done with the lock. Done.